File: nitpickers/pyql/pyql/ast/form/statement.py
from pyql.ast import ast


class Statement(ast.ASTNode):

    def __init__(self, location):
        super().__init__(location)

    def accept(self, visitor):
        return visitor.visit_statement(self)

    def __repr__(self):
        return "AST Statement at: " + str(self.location)

File: Rasha_Toine/Rasha/QLForm/src/ast/statement/Statement.java
package ast.statement;

import ast.AstNode;
import utils.CodeReference;
import visiting.StatementVisitor;

public abstract class Statement extends AstNode {

	public Statement(CodeReference location) {
		super(location);
	}
	
	public abstract <T, U> T accept(StatementVisitor<T, U> visitor, U ctx);
}